code/game/objects/items/devices/instruments.dm

```diff
@@ -205,11 +205,19 @@
 	w_class = WEIGHT_CLASS_SMALL
 	actions_types = list(/datum/action/item_action/instrument)
 
-/obj/item/instrument/harmonica/speechModification(message)
+/obj/item/instrument/harmonica/proc/handle_speech(datum/source, list/speech_args)
 	if(song.playing && ismob(loc))
 		to_chat(loc, "<span class='warning'>You stop playing the harmonica to talk...</span>")
 		song.playing = FALSE
-	return message
+
+/obj/item/instrument/harmonica/equipped(mob/M, slot)
+	RegisterSignal(M, COMSIG_MOB_SAY, .proc/handle_speech)
+	return ..()
+
+/obj/item/instrument/harmonica/dropped(mob/M)
+	UnregisterSignal(M, COMSIG_MOB_SAY)
+	return ..()

```

Comment:
Please prefer `. = ..()` as first line, a lot of your stuff in this pr doesn't use the parent call like it should